### PR TITLE
Give `Store::queryDerivationOutputMap` and `evalStore` argument

### PR DESCRIPTION
**Pull request by Ericson2314**
# Motivation

Picking up where https://github.com/NixOS/nix/pull/9563 left off.

# Context

# Priorities

Add :+1: to [pull requests you find important](https://github.com/NixOS/nix/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc).

### CODE REVIEW

**Comment by Ericson2314** [comment]
N.B. https://github.com/thufschmitt/hydra/pull/2 Fixed up Hydra CA supporting using (a backport of) this.

**Comment by roberth** [comment]
Needs tests.

The best approach may be to provide a mode to run the test suite with a separate eval store.

File: src/libstore/build/derivation-goal.cc [hunk 1]
```diff
@@ -604,7 +604,7 @@ void DerivationGoal::inputsRealised()
                     return *outPath;
                 }
                 else {
-                    auto outMap = worker.evalStore.queryDerivationOutputMap(depDrvPath);
```
Comment (roberth) [inline src/libstore/build/derivation-goal.cc:607]: Small change to CA behavior. Ignores `store` from now on.
This is in an experimental feature, relating to an already known-broken non-experimental feature.
@Ericson2314 continues to work on this.

**Review by roberth — APPROVED** [review]
No changes to non-experimental behavior.
Merging to unblock other work, which will include tests, such as #9589.

**Review by roberth — APPROVED** [review]
No behavioral change. Will be tested as mentioned.
Merging to unblock other work.

**Pull request merged**